docker/docker.ts

```diff
@@ -175,36 +180,76 @@ async function buildAndPushImageWorkerAsync(
     };
 
     // If the container specified a cacheFrom parameter, first set up the cached stages.
-    let cacheFrom: Promise<string[] | undefined>;
+    let cacheFrom: string[] | undefined;
     if (typeof pathOrBuild !== "string" && pathOrBuild && pathOrBuild.cacheFrom) {
-        // NOTE: we pull the promise out of the repository URL s.t. we can observe whether or not it exists. Were we
-        // to instead hang an apply off of the raw Input<>, we would never end up running the pull if the repository
-        // had not yet been created.
+        // NOTE: we pull the promise out of the repository URL s.t. we can observe whether or not it
+        // exists. Were we to instead hang an apply off of the raw Input<>, we would never end up
+        // running the pull if the repository had not yet been created.
         const cacheFromParam = typeof pathOrBuild.cacheFrom === "boolean" ? {} : pathOrBuild.cacheFrom;
-        cacheFrom = pullCacheAsync(imageName, cacheFromParam, login, repositoryUrl, logResource);
-    } else {
-        cacheFrom = Promise.resolve(undefined);
+        cacheFrom = await pullCacheAsync(imageName, cacheFromParam, login, repositoryUrl, logResource);
```

Comment:
Did we have to change this?  I'm a little nervous that this subtly changes some execution behaviour that we depended on. (We now block on the `pullCacheAsync` instead of running it in parallel with other work below?).  Given that we don't have much test coverage around the `cacheFrom` support, I'm inclined to not change it unless we have to - and if we do, to ideally add some additional test coverage.

Reply:
We do not.  I'll revert this.  It seemed like this existed at a time when we didn't have async/await here and were instead using Outputs.  We cleaned that up at one point, but this stuff still remained.  changing this made things clearer for me.  However, you're correct this isn't necessary, so i'll revert :)